ee/localserver/krypto-ec-middleware.go

```diff
@@ -459,3 +366,369 @@ func (bhr *bufferedHttpResponse) WriteHeader(code int) {
 func (bhr *bufferedHttpResponse) Bytes() []byte {
 	return bhr.buf.Bytes()
 }
+
+// detectPresence prompts the user for presence detection and sends periodic status updates to
+// k2 (waiting , complete, error, timeout)
+func (e *kryptoEcMiddleware) detectPresence(challengeBox *challenge.OuterChallenge) {
+	// extract cmd req from challenge box
+	ctx, cancel := context.WithTimeout(context.Background(), presencedetection.DetectionTimeout)
+	defer cancel()
+
+	var cmdReq v2CmdRequestType
+	if err := json.Unmarshal(challengeBox.RequestData(), &cmdReq); err != nil {
+		e.slogger.Log(ctx, slog.LevelError,
+			"error unmarshaling cmd request",
+			"err", err,
+		)
+		return
+	}
+
+	// figure out if we need to do presence detection
+	detectionIntervalStr, ok := cmdReq.Headers[kolidePresenceDetectionIntervalHeaderKey]
+	if !ok || len(detectionIntervalStr) == 0 || detectionIntervalStr[0] == "" {
+		return
+	}
+
+	// presence detection is not yet available on linux
+	if runtime.GOOS == "linux" {
```

Comment:
nit- this check is cheap and doesn't depend on any of the other work done, maybe worth having earlier in the method